.github/workflows/build.yml

```diff
@@ -71,6 +71,7 @@ jobs:
             code_coverage: false
             static: true
           - os: macos15
+            conan_profile: default_apple_clang
```

Comment:
This change serves several purposes:
1. If default clang version changes, the profile name will still be correct.
2. If we change the runner, it will also stay correct
3. We don't have to patch the name for conan profile in our action